dp/cloud/python/magma/configuration_controller/response_processor/strategies/response_processing.py

```diff
@@ -202,30 +209,33 @@ def process_heartbeat_response(obj: ResponseDBProcessor, response: DBResponse, s
     Returns:
         None
     """
+    grant = _get_grant_from_response(response=response, session=session)
 
-    grant = _get_or_create_grant_from_response(obj, response, session)
-    if not grant:
+    if response.response_code == ResponseCodes.TERMINATED_GRANT.value:
+        if grant:
+            unset_frequency(grant=grant)
+            session.delete(instance=grant)
```

Comment:
Why sometimes there is `session.delete(instance=grant)` and the other times it is `session.delete(grant)` (I prefer the second one)

Reply:
I'm fan of Python's keyword args due to readability, but you're right I should've matched preexisting code convention for the given repo.   
Switched to positional args.